src/drivers/amd/rembrandt/acp_sw_audio_dma.c

```diff
@@ -0,0 +1,640 @@
+{
+	switch (type) {
+	case DMA_ATTR_BUFFER_ALIGNMENT:
+	case DMA_ATTR_COPY_ALIGNMENT:
+		*value = ACP_DMA_BUFFER_ALIGN_128;
+		break;
+	case DMA_ATTR_BUFFER_ADDRESS_ALIGNMENT:
+		*value = PLATFORM_DCACHE_ALIGN;
+		break;
+	case DMA_ATTR_BUFFER_PERIOD_COUNT:
+		*value = ACP_DAI_DMA_BUFFER_PERIOD_COUNT;
+		break;
+	default:
+		return -ENOENT;
+	}
+	return 0;
+}
+
+static int acp_dai_sw_audio_dma_interrupt(struct dma_chan_data *channel, enum dma_irq_cmd cmd)
+{
+	uint32_t status = 0;
+	acp_dsp0_intr_stat_t acp_intr_stat;
+	acp_dsp0_intr_cntl_t acp_intr_cntl;
+	acp_dsp0_intr_stat1_t acp_intr_stat1;
+	acp_dsp0_intr_cntl1_t acp_intr_cntl1;
+
+	if (channel->status == COMP_STATE_INIT)
+		return 0;
+	switch (cmd) {
+	case DMA_IRQ_STATUS_GET:
+		if (channel->index == SDW1_ACP_P1_SW_BT_TX_EN_CH ||
+		    channel->index == SDW1_ACP_P1_SW_BT_RX_EN_CH) {
+			acp_intr_stat1 =  (acp_dsp0_intr_stat1_t)dma_reg_read(channel->dma,
+			sw_dev[channel->index].sw_dev_dma_intr_status);
+			status = acp_intr_stat1.bits.audio_buffer_int_stat;
+		} else {
+			acp_intr_stat =  (acp_dsp0_intr_stat_t)dma_reg_read(channel->dma,
+			sw_dev[channel->index].sw_dev_dma_intr_status);
+			status = acp_intr_stat.bits.audio_buffer_int_stat;
+		}
+		return (status & (1 << sw_dev[channel->index].statusindex));
+	case DMA_IRQ_CLEAR:
+		if (channel->index == SDW1_ACP_P1_SW_BT_TX_EN_CH ||
+		    channel->index == SDW1_ACP_P1_SW_BT_RX_EN_CH) {
+			acp_intr_stat1.u32all = 0;
+			acp_intr_stat1.bits.audio_buffer_int_stat =
+						(1 << sw_dev[channel->index].statusindex);
+			status  = acp_intr_stat1.u32all;
+			dma_reg_write(channel->dma, sw_dev[channel->index].sw_dev_dma_intr_status,
+				      status);
+		} else {
+			acp_intr_stat.u32all = 0;
+			acp_intr_stat.bits.audio_buffer_int_stat =
+						(1 << sw_dev[channel->index].statusindex);
+			status  = acp_intr_stat.u32all;
+			dma_reg_write(channel->dma, sw_dev[channel->index].sw_dev_dma_intr_status,
+				      status);
+		}
+		return 0;
+	case DMA_IRQ_MASK:
+		if (channel->index == SDW1_ACP_P1_SW_BT_TX_EN_CH ||
+		    channel->index == SDW1_ACP_P1_SW_BT_RX_EN_CH) {
+			acp_intr_cntl1 =  (acp_dsp0_intr_cntl1_t)dma_reg_read(channel->dma,
+			sw_dev[channel->index].sw_dev_dma_intr_cntl);
+			acp_intr_cntl1.bits.audio_buffer_int_mask &=
+			(~(1 << sw_dev[channel->index].statusindex));
+			status  = acp_intr_cntl1.u32all;
+			dma_reg_write(channel->dma, sw_dev[channel->index].sw_dev_dma_intr_cntl,
+				      status);
+		} else {
+			acp_intr_cntl =  (acp_dsp0_intr_cntl_t)
+			io_reg_read(PU_REGISTER_BASE + sw_dev[channel->index].sw_dev_dma_intr_cntl);
+			acp_intr_cntl.bits.audio_buffer_int_mask &=
+			(~(1 << sw_dev[channel->index].statusindex));
+			status  = acp_intr_cntl.u32all;
+			dma_reg_write(channel->dma, sw_dev[channel->index].sw_dev_dma_intr_cntl,
+				      status);
+		}
+		return 0;
+	case DMA_IRQ_UNMASK:
+		if (channel->index == SDW1_ACP_P1_SW_BT_TX_EN_CH ||
+		    channel->index == SDW1_ACP_P1_SW_BT_RX_EN_CH) {
```

Comment:
could be a `switch (channel->index)` too

Reply:
done